client/my-sites/media/controller.js

```diff
@@ -31,4 +32,14 @@ export default {
 		} );
 		next();
 	},
+	singleMedia: function( context, next ) {
+		if ( ! getSiteFragment( context.path ) ) {
+			return page.redirect( '/media' );
+		}
+		// Render
+		context.primary = React.createElement( SingleMediaComponent, {
```

Comment:
Do we need to use `React.createElement` here? Using something simpler like:

```
context.primary = <SingleMediaComponent attachment={ context.params.attachment } />;
```

should do the job as well.